### DISCUSSION TITLE
how can gluonts deal with NaN values in Features of Training data

### DISCUSSION THREAD

**Discussion by pratikgehlott** [awslabs/gluonts#1533] [Q&A] [answered]
I need to know how can we impute or do something to NaN values so that deepAR estimator can accept it.

**Comment by mbohlkeschneider** [comment, marked as answer]
Hi @Pratik325, 

`NaN` values are not supported by DeepAR for features at the moment.

**Reply by pratikgehlott** [reply]
okay thanks @mbohlkeschneider

**Reply by pratikgehlott** [reply]
So @mbohlkeschneider  how can we deal with missing dates in the data, for example if we dont have data from particular date, how can we feed that into the model?

**Reply by mbohlkeschneider** [reply]
You would have to impute that data before passing it to the model.

**Reply by pratikgehlott** [reply]
got it thanks @mbohlkeschneider, I wonder if gluons had some internal parameter to do imputation, that would be great.

**Reply by mbohlkeschneider** [reply]
It does have for the target but not for the features at the moment.

**Reply by pratikgehlott** [reply]
okay got it. Thanks @mbohlkeschneider

**Reply by mbohlkeschneider** [reply]
One think you could try is adding another transformation to impute the values after the features are vstacked: 

https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/model/deepar/_estimator.py#L342

But I have never tried that.

**Reply by pratikgehlott** [reply]
okay @mbohlkeschneider
